### ISSUE TITLE
False positive: calling super on bases base class is shown as error

### ISSUE THREAD

**Issue by Bergiu** [PyCQA/pylint#2903]
### Steps to reproduce
1. Make an empty folder and create the file test.py with this content:
![image](https://user-images.githubusercontent.com/13796963/57178525-daa33680-6e71-11e9-94fe-a7d6f19c1c27.png)
2. `pylint test.py`

### Current behavior
Executing the code works, but a false positive error is shown.

`E1003: Bad first argument 'B' given to super() (bad-super-call)`

### Expected behavior
Pylint shouldn't show this error.

### pylint --version output
```pylint 2.3.1
astroid 2.2.5
Python 3.7.3 (default, Mar 26 2019, 21:43:19) 
[GCC 8.2.1 20181127]
```

**Comment by PCManticore** [comment]
@Bergiu At line 17 you are supposed to pass `C` to super()`, not `B`. By passing `B` you basically ask a MRO navigation starting from B instead of C for finding the `something` method.

**Comment by Bergiu** [comment]
But I want it to start at B, so that I get the `something` method from A.

**Comment by Bergiu** [comment]
Can you reopen the issue @PCManticore, or am I getting something wrong?

**Comment by maxhillis** [comment]
What is the difference between lines 15 and 17? Why is 15 ok and 17 bad?

**Comment by Bergiu** [comment]
the method is called in line 17.

**Comment by maxhillis** [comment]
No sorry I mean from the perspective of pylint. If 17 is flagged then 15 should be too. I was trying to do the same thing you were and I couldn't understand why pylint considered this bad code. It works as I intend but I have been having trouble finding why its considered bad practice

**Comment by Bergiu** [comment]
Yes

**Comment by krystean** [comment]
Came across this while making a project which has the same structure as your code and can't understand why it's considered a bad-super-call. Isn't that the use of super?

**Comment by DanielNoord** [comment]
Re-opening this because we found ourselves that it is probably better to do consider this a false positive. If anybody feels like implementing this I think we would be more than happy to review and put this in one of the next releases!

For reference see the small discussion in:
https://github.com/PyCQA/astroid/pull/1217#discussion_r744149027

**Comment by cdce8p** [comment]
For reference and testing:
```py
class Parent:
    def __init__(self):
        print("Parent")

class Child(Parent):
    def __init__(self):
        print("Child")
        super().__init__()

class GrandChild(Child):
    def __init__(self):
        print("Grandchild")
        super(Child, self).__init__()  # should not emit 'bad-super-call'!

GrandChild()
```

**Comment by cdce8p** [comment]
For reference: discussion about future usefulness of check: https://github.com/PyCQA/astroid/pull/1217#discussion_r744170384
